server/utils/passkey/session.ts

```diff
@@ -0,0 +1,221 @@
+import { UserProfile } from '@prisma/client'
+import jwt from 'jsonwebtoken'
+
+const jwtSecret = process.env.SUPABASE_JWT_SECRET || ''
+const jwtIssuer = process.env.SUPABASE_URL || ''
+
+/**
+ * Creates an access token for WebAuthn authentication that works with Supabase Auth.
+ * 
+ * This token is compatible with the custom_access_token_hook function that adds additional
+ * session metadata to the JWT claims.
+ * 
+ * @param user The user profile
+ * @param sessionData Optional session data to include in the token
+ */
+export function createWebAuthnAccessTokenForUser(
+  user: UserProfile, 
+  sessionData?: {
+    id?: string;
+    deviceNonce?: string;
+    ip?: string;
+    userAgent?: string;
+    createdAt?: Date;
+    updatedAt?: Date;
+    applicationIds?: string[];
+  }
+) {
+  const issuedAt = Math.floor(Date.now() / 1000)
+  const expirationTime = issuedAt + 3600 // 1 hour expiry
+  
+  // Create session data for app_metadata
+  const sessionMetadata = sessionData ? {
+    sessionData: {
+      ip: sessionData.ip || '',
+      userAgent: sessionData.userAgent || '',
+      deviceNonce: sessionData.deviceNonce || '',
+      createdAt: sessionData.createdAt ? sessionData.createdAt.toISOString() : '',
+      updatedAt: sessionData.updatedAt ? sessionData.updatedAt.toISOString() : '',
+      applicationIds: sessionData.applicationIds || []
+    }
+  } : {};
+  
+  // Define the payload type with optional session_id
+  type JWTPayload = {
+    iss: string;
+    sub: string;
+    aud: string;
+    exp: number;
+    iat: number;
+    email: string | null;
+    phone: string | null;
+    app_metadata: {
+      sessionData?: {
```

Comment:
Session data seems to be at the root, not nested inside `app_metadata`. `app_metadata` could be used to identify this sessions as originating from a passkeys authentication, as it usually contains something like this:

```
"app_metadata": {
    "provider": "twitter",
    "providers": [
      "twitter"
    ]
  }
```

However, if you fetch a session on the client, you'll `session.user` also contains an `identities` property. I'm not sure if manually setting the value of `provider` and `providers` will throw any error, as it's related to this: https://supabase.com/docs/guides/auth/auth-identity-linking